### DISCUSSION TITLE
How to insert json into postgre?

### DISCUSSION THREAD

**Discussion by wuzhen1228** [prest/prest#816] [API Server] [answered]
my json

```json
{
    "username": "admin",
    "topic": "asdf",
    "pub_props" : {"age":"22","sex",0}
}
```

In the body, `pub_props` is nested json
prest prompt `"sql: converting argument $7 type: unsupported type map[string]interface {}, a map"`

How to solve it?

**Comment by felipeweb** [comment]
@wuzhen1228  Could you share your table structure?

**Comment by avelino** [comment, marked as answer]
@wuzhen1228 
in your json has a typo, was it a typo `,` or are you using it this way in the request?

`"pub_props" : {"age":"22","sex",0}`

from: `"sex",0`
to: `"sex":0`

**Comment by wuzhen1228** [comment]
> 您的 json 中的@wuzhen1228有错字，是错字`,`还是您在请求中以这种方式使用它？
> 
> `"pub_props" : {"age":"22","sex",0}`
> 
> 从：`"sex",0`
> 到：`"sex":0`

yes，but it still prompts an error

**Comment by wuzhen1228** [comment]
> @wuzhen1228 Could you share your table structure?

The type of the field is jsonb

**Comment by wuzhen1228** [comment]
> @wuzhen1228 Could you share your table structure?

The type of the field is jsonb

**Comment by toplinuxsir** [comment]
@avelino , not supported jsonb filed insert ?
Thanks!

**Reply by avelino** [reply]
prest has jsonb support

https://docs.prestd.com/prestd/docs/api-reference/advanced-queries/

I use jsonb in several projects
